latch/ldata/path.py

```diff
@@ -0,0 +1,271 @@
+        """
+        data = execute(
+            gql.gql("""
+            query LDataChildren($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        type
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise FileNotFoundError(f"No such Latch file or directory: {self.path}")
+        if data["finalLinkTarget"]["type"].lower() not in dir_types:
+            raise ValueError(f"{self.path} is not a directory")
+
+        for node in data["finalLinkTarget"]["childLdataTreeEdges"]["nodes"]:
+            yield LPath(urljoins(self.path, node["child"]["name"]))
+
+    def rmr(self) -> None:
+        """Recursively delete files at this instance's path.
+
+        Always makes a network request.
+        """
+        execute(
+            gql.gql("""
+            mutation LDataRmr($nodeId: BigInt!) {
+                ldataRmr(input: { argNodeId: $nodeId }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id},
+        )
+
+    def copy_to(self, dst: "LPath", *, show_summary: bool = False) -> None:
+        _remote_copy(self.path, dst.path, show_summary=show_summary)
+
+    def upload_from(self, src: Path, *, show_progress_bar: bool = False) -> None:
+        _upload(
+            os.fspath(src),
+            self.path,
+            progress=Progress.tasks if show_progress_bar else Progress.none,
+            verbose=False,
+        )
+
+    def download(
+        self, dst: Optional[Path] = None, *, show_progress_bar: bool = False
+    ) -> Path:
+        if dst is None:
+            global download_idx
+            dir = Path.home() / "lpath"
+            dir.mkdir(parents=True, exist_ok=True)
+            dst = dir / f"{download_idx}_{self.name()}"
+            download_idx += 1
+
+        _download(
+            self.path,
+            dst,
+            progress=Progress.tasks if show_progress_bar else Progress.none,
+            verbose=False,
+            confirm_overwrite=False,
+        )
+        return dst
+
+    def __truediv__(self, other: object) -> "LPath":
+        if not isinstance(other, (LPath, str)):
+            return NotImplemented
+        return LPath(urljoins(self.path, other))
+
+
+class LPathTransformer(TypeTransformer[LPath]):
+    _TYPE_INFO = BlobType(
+        # rahul: there is no way to know if the LPath is a file or directory
+        # ahead to time, so just set dimensionality to SINGLE
+        format="binary",
```

Comment:
```suggestion
        format="",
```
format is pretty much always empty in flyte I think